geode-core/src/test/java/org/apache/geode/management/internal/cli/commands/ImportDataIntegrationTest.java

```diff
@@ -0,0 +1,158 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more contributor license
+ * agreements. See the NOTICE file distributed with this work for additional information regarding
+ * copyright ownership. The ASF licenses this file to You under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ *
+ */
+
+package org.apache.geode.management.internal.cli.commands;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.Assert.assertEquals;
+
+import java.util.stream.IntStream;
+
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+
+import org.apache.geode.management.internal.cli.i18n.CliStrings;
+import org.apache.geode.management.internal.cli.util.CommandStringBuilder;
+import org.apache.geode.test.junit.categories.IntegrationTest;
+
+@Category(IntegrationTest.class)
+public class ImportDataIntegrationTest extends SnapshotDataIntegrationTest {
```

Comment:
In general we would want to discourage usage of abstract test classes and in favor of rules. If there is minimum set up needed using the rules, we don't mind a few lines of duplicate code for set up to avoid entangling of the tests. What if in the future you want to change the setup of one test instead of another. Extending a common abstract class makes it harder to add on to the test.

Reply:
I tried to include only pretty generic code to share between the two tests, though (as with any code) it is always possible that they could diverge in the future, making the abstract class less / no longer valuable. I am not opposed to duplicating the shared code in each location instead if there is a stronger preference towards ease of test extension / modification over DRYness.